src/frontend/src/sol/services/spl.services.ts

```diff
@@ -155,32 +156,61 @@ export const loadUserTokens = async ({
 
 		const splCustomTokens = await loadSplCustomTokens({ identity, certified: true });
 
-		const contracts = await saveCachedUserTokensToBackend({
-			identity,
-			savedTokens: splCustomTokens
-		});
+		await saveCachedUserTokensToBackend({ identity, savedTokens: splCustomTokens });
 
-		const [existingTokens, userTokenAddresses] = contracts.reduce<
-			[SplUserToken[], SplTokenAddress[]]
+		const [existingTokens, nonExistingTokens] = splCustomTokens.reduce<
+			[SplUserToken[], SplUserToken[]]
 		>(
-			([accExisting, accUser], address) => {
-				const existingTokens = SPL_TOKENS.reduce<SplUserToken[]>(
-					(acc, token) =>
-						contracts.includes(token.address) ? [...acc, { ...token, enabled: true }] : acc,
-					[]
+			([accExisting, accNonExisting], { token, enabled, version: versionNullable }) => {
+				if (!('SplMainnet' in token || 'SplDevnet' in token)) {
+					return [accExisting, accNonExisting];
+				}
+
+				const version = fromNullable(versionNullable);
+
+				const {
+					network: tokenNetwork,
+					symbol,
+					decimals,
+					token_address: tokenAddress
+				} = 'SplDevnet' in token
+					? { ...token.SplDevnet, network: SOLANA_DEVNET_NETWORK }
+					: { ...token.SplMainnet, network: SOLANA_MAINNET_NETWORK };
```

Comment:
Do you want some kind of fail-safe, in case this is missed when more Spl networks are added?